cypress/integration/email-notification-form-spec.js

```diff
@@ -0,0 +1,215 @@
+        .get('input[id=agree_to_emails]').click()
+    })
+  });
+
+  it('Should be able to submit the form & see a message confirming successful subscription', () => {
+    cy.intercept('POST', 'http://localhost:3001/api/v1/users', (req) => {
+      req.reply({
+        statusCode: 200,
+        fixture: 'success.json'
+      })
+    }).as('successfulPost')
+
+    cy.fixture('user1.json').as('user1').then((user1) => {
+      cy.get('input[id=first_name]').type(user1.first_name)
+        .get('input[id=last_name]').type(user1.last_name)
+        .get('select[id=state_name]').select(user1.state_name)
+        .get('input[id=email]').type(user1.email)
+        .get('input[id=english]').click()
+        .get('input[id=agree_to_emails]').click()
+        .get('.submit-button').click()
+        .wait('@successfulPost')
+        .get('.server-message').should('contain', 'You are now registered to receive notifications about upcoming elections in your state.')
+    })
+  });
+
+  it('Should be given a list of fifty states upon selecting the dropdown menu', () => {
+    cy.get('#state_name option').then(options => {
+      const actual = [...options].map(option => option.value);
+      cy.fixture('states.json').as('states').then((states) => {
+        const statesValue = states.map(state => state.value)
+        expect(actual).to.deep.eq(statesValue)
+      })
+    })
+  });
+
+  xit('Should be able to navigate & submit the form using only the keyboard', () => {
+    cy.intercept('POST', 'http://localhost:3001/api/v1/users', (req) => {
+        req.reply({
+          fixture: 'success.json'
+        })
+      }).as('sucessfulPost')
+
+    cy.fixture('user2.json').as('user2').then((user2) => {
+      cy.get('form')
+        .realPress('Tab').realPress('Tab').realPress('Tab')
+        .focused().should('have.id', 'first_name')
+        .type(user2.first_name)
+        .realPress('Tab')
+        .focused().should('have.id', 'last_name')
+        .type(user2.last_name)
+        .realPress('Tab')
+        .focused().should('have.id', 'state_name')
+        .realType(`${user2.state_name}`)
+        .realPress('Tab')
+        .focused().should('have.id', 'email')
+        .type(user2.email)
+        .realPress('Tab')
+        .focused().should('have.id', 'english')
+        .realPress('Space')
+        .get(`input[value=${user2.language}]`).should('have.value', user2.language)
+        .realPress('Tab')
+        .focused().should('have.id', 'agree_to_emails')
+        .get('input[id=agree_to_emails]').should('have.value', 'false')
+        .realPress('Space')
+        .get('input[id=agree_to_emails]').should('have.value', 'true')
+        .realPress('Tab')
+        .realPress('Enter')
+        .wait('@successfulPost')
+        .get('.server-message').should('contain', 'You are now registered to receive notifications about upcoming elections in your state.')
+    })
+  });
```

Comment:
I am going to get this to work gosh darn it!  It's funny, it actually works about half of the time.  When Cypress gets to the `state_name` select, it almost seems like it doesn't wait long enough to get 'co'.  I tried adding a `cy.wait` at some point, but things changed after that.  Maybe I will try it again...

Reply:
We can always come back and add more tests like this one. For now, I think it would be good to remove it since we don't want it to interfere with our ability to implement Continuous Integration and Deployment.

Reply:
Got it!

![](https://files.slack.com/files-pri/T029P2S9M-F032KUXF7QE/screen_shot_2022-02-11_at_13.25.59.png)